src/components/Button/ButtonContainer.js

```diff
@@ -0,0 +1,171 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import styled, { css } from 'react-emotion';
+
+import { omit } from '../../util/fp';
+import { sizes } from '../../styles/constants';
+import { Button, LoadingIcon } from './components';
+import {
+  LOADING_STATES,
+  EXIT_ANIMATION_DURATION,
+  SIZE_PROP_TYPE
+} from './constants';
+
+const { KILO, MEGA, GIGA } = sizes;
+
+const ContentWrapper = styled('div')`
+  position: relative;
+`;
+
+// TODO: use theme animations here.
+const childrenWrapperStyles = () => css`
+  opacity: 1;
+  transform: translate3d(0, 0, 0);
+  transition: opacity 200ms ease-in-out, transform 200ms ease-in-out;
+`;
+
+const childrenWrapperLoadingStyles = ({ loadingState }) =>
+  loadingState !== LOADING_STATES.INACTIVE &&
+  css`
+    opacity: 0;
+    transform: translate3d(0, 50px, 0);
+  `;
+
+const ChildrenWrapper = styled('div')`
+  ${childrenWrapperStyles};
+  ${childrenWrapperLoadingStyles};
+`;
+
+/**
+ * @component
+ */
+export default class ButtonContainer extends Component {
+  static KILO = KILO;
+  static MEGA = MEGA;
+  static GIGA = GIGA;
+  static propTypes = {
+    /**
+     * Children nodes. Automatically wrapped in a div, so you can pass multiple
+     * elements.
+     */
+    children: PropTypes.node.isRequired,
+    /**
+     * Standard onClick function. If used on an anchor this can be used to
+     * cause additional side-effects like tracking.
+     */
+    onClick: PropTypes.func,
+    /**
+     * Callback for when the button's success/error animations are complete.
+     * Receives the original promise returned from onClick.
+     */
+    onAnimationComplete: PropTypes.func,
+    /**
+     * Delay to wait before showing loading state.
+     */
+    delayMs: PropTypes.number,
+    /**
+     * Size of the button. Use the Button's KILO, MEGA, or GIGA properties.
+     */
+    size: SIZE_PROP_TYPE
+  };
+
+  static defaultProps = {
+    onClick: null,
+    delayMs: 500,
+    size: MEGA,
+    onAnimationComplete: null
+  };
+
+  state = {
+    timeoutId: null,
+    loadingState: LOADING_STATES.INACTIVE
+  };
+
+  componentWillUnmount() {
+    const { timeoutId } = this.state;
+    if (timeoutId) {
+      clearTimeout(timeoutId);
+    }
+  }
+
+  handleClick = e => {
```

Comment:
I think this is good for the first iteration. Let's try using it and see how it goes